components/suggestion/suggestion_box/use_suggestion_handler.ts

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useMemo} from 'react';
```

Comment:
I like having separate handlers it makes it readable and testable, I wonder if we can go even further in detecting unnecessary functionality and stripping it?

Reply:
Well this can be solved.